### PR TITLE
Drop support for Node < 10

### LINKED ISSUE #930: Drop support for Node versions < 10
Node v4's end-of-life was 2018-04-30 (ref https://github.com/nodejs/Release#end-of-life-releases).

Can we drop support for it? cc @springmeyer

### PR DESCRIPTION
**Pull request by jseppi**
Closes #930

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -166,14 +113,14 @@ matrix:
...
-      node_js: 4
+      node_js: 10
       # Overrides `script` to disable publishing
       before_script:
         - npm test
     # Coverage build
     - os: linux
       env: BUILDTYPE=debug CXXFLAGS="--coverage" LDFLAGS="--coverage"
-      node_js: 4
+      node_js: 10
```
Comment (artemp) [inline .travis.yml]: I'm wondering if we should upgrade GCC version as well ? /cc @springmeyer

File: .travis.yml [hunk 2]
```diff
@@ -144,7 +91,7 @@ matrix:
     # g++ build (default builds all use clang++)
     - os: linux
       env: BUILDTYPE=debug CXX="g++-6" CC="gcc-6"
-      node_js: 4
+      node_js: 10
```
Comment (artemp) [inline .travis.yml]: I'm wondering if we should upgrade GCC version as well ? /cc @springmeyer

Comment (springmeyer) [reply .travis.yml]: No strong preference on gcc. This was added to get access to the `-Weffc++` warnings mainly.

Comment (jseppi) [reply .travis.yml]: @artemp feel free to upgrade the GCC version in this PR if you'd like.
I don't have much experience there, so I'd like to leave it for you to use the best version :)

**Pull request merged**